src/main/java/pixelperfect/route/RouteGenerator.java

```diff
@@ -4,8 +4,36 @@
  * Random generator for routes through space.
  * 
  * @author Jesse Tilro
+ * @author Floris Doolaard
  *
  */
 public class RouteGenerator {
 
+	private Route route;
```

Comment:
This way, the RouteGenerator can only generate a single route. I would model it to be just a machine that generate a new route each time it is requested to do so.